workflows/e2e-test.yml

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Canonical Ltd.
+name: E2E tests
+
+on:
+  workflow_call:
+    inputs:
+      branch-name:
+        description: Name of the branch to checkout
+        required: false
+        type: string
+        default: ${{ github.ref }}
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: docker.io
+      DOCKER_REGISTRY: docker.io/
+      DOCKER_REPOSITORY: omectesting/
+    steps:
+      - uses: actions/checkout@v4
+      - uses: docker/login-action@v3.3.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: "omectesting"
+          password: "dckr_pat_Up1vh3lGEKIfGDJuEj7Qb7OfdEE"
```

Comment:
Shouldn't this be a secret?

This particular password should also be changed and not put as is in a secret.